src/did.rs

```diff
@@ -405,6 +502,36 @@ pub trait DIDMethod: DIDResolver {
         None
     }
 
+    /// Retrieve a DID from a DID method transaction
+    fn did_from_transaction(&self, _tx: DIDMethodTransaction) -> AResult<String> {
+        bail!("DID from transaction not implemented for DID Method");
+    }
+
+    /// Submit a DID transaction
+    async fn submit_transaction(&self, _tx: DIDMethodTransaction) -> AResult<Value> {
+        bail!("Transaction submission not implemented for DID Method");
+    }
+
+    /// Create a DID
+    fn create(&self, _create: DIDCreate) -> AResult<DIDMethodTransaction> {
+        bail!("Create operation not implemented for DID Method");
+    }
+
+    /// Update a DID
+    fn update(&self, _update: DIDUpdate) -> AResult<DIDMethodTransaction> {
+        bail!("Update operation not implemented for DID Method");
+    }
+
+    /// Recover a DID
+    fn recover(&self, _recover: DIDRecover) -> AResult<DIDMethodTransaction> {
+        bail!("Recover operation not implemented for DID Method");
+    }
+
+    /// Deactivate a DID
+    fn deactivate(&self, _deactivate: DIDDeactivate) -> AResult<DIDMethodTransaction> {
+        bail!("Deactivate operation not implemented for DID Method");
+    }
+
```

Comment:
Similar to https://github.com/spruceid/ssi/pull/379/files#r806777763 about the `Sidetree` trait..
I since learned that `anyhow` is not recommended for APIs; maybe these should be changed back to using `ssi::error::Error` or something? Error associated type seems problematic for object-safety (also discussed in https://github.com/spruceid/ssi/pull/379/files#r807305656 about DID method transactions).

Reply:
Opened https://github.com/spruceid/ssi/issues/392

Reply:
Wouldn't it be preferable to do the switch now to avoid breaking the API in the future? Not that it's a lot of work to adapt code for consumers.

The code could simply be
```rust
#[derive(Error, Debug)]
pub enum MyError {
    #[error(transparent)]
    Other(#[from] anyhow::Error),  // source and Display delegate to anyhow::Error
}
```

Reply:
Trying that... It seems that `bail!` and `ensure!` cannot be used then.

Reply:
Done in a7879ce22a5eee6ca7ae66a201915892d29945c2